plugins/common/cookie/cookie.go

```diff
@@ -0,0 +1,91 @@
+package cookie
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+type CookieAuthConfig struct {
+	URL    string `toml:"cookie_auth_url"`
+	Method string `toml:"cookie_auth_method"`
+
+	// HTTP Basic Auth Credentials
+	Username string `toml:"cookie_auth_username"`
+	Password string `toml:"cookie_auth_password"`
+
+	Body    string          `toml:"cookie_auth_body"`
+	Renewal config.Duration `toml:"cookie_auth_renewal"`
+
+	client *http.Client
+}
+
+func (c *CookieAuthConfig) Start(client *http.Client) (err error) {
```

Comment:
Hmm, is there a reason for passing in a client? You could also setup the client here...

Reply:
The `client` has to have the same `cookiejar`, so might as well use the same `client`.

Reply:
Understood. Maybe you can add your comment above as a small comment to the function...